src/test/java/spoon/test/imports/ImportTest.java

```diff
@@ -403,8 +403,8 @@ public void testNestedAccessPathWithTypedParameter() throws Exception {
 		}
 		CtClass<?> mm = launcher.getFactory().Class().get("spoon.test.imports.testclasses2.AbstractMapBasedMultimap");
 		CtClass<?> mmwli = launcher.getFactory().Class().get("spoon.test.imports.testclasses2.AbstractMapBasedMultimap$WrappedList$WrappedListIterator");
-		assertTrue(mmwli.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedList.WrappedIterator")>=0);
-		assertTrue(mm.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedList.WrappedIterator")>=0);
+		assertTrue(mmwli.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedCollection.WrappedIterator")>=0);
+		assertTrue(mm.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedCollection.WrappedIterator")>=0);
```

Comment:
It could be great to add the test: we check with @tdurieux and it will work as defined in the test. However if you build the model and then you put dynamically a new class `WrappedIterator` in `WrappedList` the `WrappedListIterator` continue to extend the old `WrappedIterator` and not the newly one.
There is a real discussion here on what should be the behaviour, but I think it's out of the scope of this PR.